src/compas/scene/scene.py

```diff
@@ -108,19 +113,49 @@ def add(self, item, parent=None, **kwargs):
             The scene object associated with the item.
         """
 
-        parent = parent or self.root
+        if "context" in kwargs:
+            if kwargs["context"] != self.context:
+                raise Exception("Object context should be the same as scene context: {} != {}".format(kwargs["context"], self.context))
+            del kwargs["context"]  # otherwist the SceneObject receives "context" twice, which results in an error
+
+        # Create a corresponding new scene object
+        sceneobject = SceneObjectFactory.create(item=item, context=self.context, scene=self, **kwargs)
```

Comment:
since this class has only 1 static method, perhaps if might as well just be a function? using a class seems overkill...

Reply:
yep that make sense